Add Ferdi server to hosted services

SpecialAro opened the pull request:
<!-- Thank you for your Pull Request. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!-- Please start by naming your pull request properly for e.g. "Add Google Tasks to Todo providers". -->
<!-- Please keep in mind that any text inside "<!--" and "--\>" are comments from us and won't be visible in your bug report, so please don't put any text in them. -->

#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/ferdium/ferdium-recipes/issues)
2. If you are requesting support for a **new service** in Ferdium, please log it [here](https://github.com/ferdium/ferdium-recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/ferdium/ferdium-app#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdium, but to get new recipes between Ferdium releases, this documentation is quite useful.)
4. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
Add Ferdi API Server to hosted services.

#### Motivation and Context
This fixes the following open issue https://github.com/ferdium/ferdium-app/issues/38.

#### Screenshots
![image](https://user-images.githubusercontent.com/37463445/165298739-a1d67857-69f6-4935-9e14-30a18343cdaf.png)

#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes
Add Ferdi API to Ferdium hosted services.

SpecialAro committed:
Add Ferdi server to hosted services

vraravam commented:
I see a lot of hits in `Login.js`, `AccountDashboard.js` and other files if I search for `LIVE_FRANZ_API` - don't they also need to accommodate the new constant?

SpecialAro commented:
Maybe. I can try to give it a look for the rest, although I'm not entirely sure they will be needed.

I tested the current code with Account Creation and Login to the ferdi API and it worked, at least.

vraravam commented:
please also ensure that all paths of code that use the `apiBase.js` are also tested. for eg, `Help > Import/Export Configuration Data` opens the browser in specific url

SpecialAro committed:
Fix the rest of LIVE_FERDI_API calls

SettingsNavigation.js and TeamDashboard.js also have a call to LIVE_FERDI_API but it handles the feature "Teams" that I think neither Ferdi or Ferdium provide.
Removes comment on apiBase.ts because I tested and the Terms where redirecting fine on the Help Menu

vraravam commented on src/api/apiBase.ts:
```diff
@@ -34,7 +34,6 @@ const apiBase = (withVersion = true) => {
 
 export function termsBase() {
-  // TODO: This needs to handle local vs ferdium vs franz servers
   return (window as any).ferdium.stores.settings.all.app.server !== LIVE_FRANZ_API
```
Do you think this if condition itself can be removed?

SpecialAro replied:
I don't think so:

-  it handles a difference on the URLs from https://api.getferdi.com or https://api.ferdium.org, where we store our terms file (in /terms)
- The DEV_API_FRANZ_WEBSITE is https://meetfranz.com

vraravam replied:
ok  - i havent delved too deep into this right now (have to start work meetings). I will resolve this comment. I also haven't run this locally, if you are confident, please squash and merge

SpecialAro replied:
I think I don't have permission to squash and merge directly on this repo 🥲
